### PR TITLE
add send feedback function for milestones

### PR DESCRIPTION
**Pull request by VandanDaolens**
Task: https://app.clickup.com/t/85zrhcngv

### CODE REVIEW

File: contracts/ApplicationRegistry.sol [hunk 1]
```diff
@@ -282,9 +284,36 @@ contract ApplicationRegistry is Ownable,Pausable,IApplicationRegistry {
...
+        uint256 _applicationId,
+        uint256 _milestoneId,
+        uint256 _workspaceId,
+        address _grantAddress,
+        string memory _feedbackMetadataHash
+    ) external onlyGrantAdminOrReviewer(_grantAddress) {
+        Application storage application = applications[_applicationId];
+        require(application.workspaceId == _workspaceId, "ApplicationStateUpdate: Invalid workspace");
+        require(application.state == ApplicationState.Approved, "MilestoneStateUpdate: Invalid application state");
+        require(_milestoneId < application.milestoneCount, "MilestoneStateUpdate: Invalid milestone id");
```
Comment (slice-amandata) [inline contracts/ApplicationRegistry.sol]: _milestoneId >=
 0 @VandanDaolens

File: contracts/ApplicationRegistry.sol [hunk 2]
```diff
@@ -282,9 +284,36 @@ contract ApplicationRegistry is Ownable,Pausable,IApplicationRegistry {
...
+        MilestoneState currentState = applicationMilestones[_applicationId][_milestoneId].state;
```
Comment (slice-amandata) [inline contracts/ApplicationRegistry.sol]: store it in memory
@VandanDaolens

**Review by slice-amandata — CHANGES_REQUESTED** [review]

File: contracts/ApplicationRegistry.sol (hunk 2 above)
Comment (VandanDaolens) [reply contracts/ApplicationRegistry.sol]: Adding memory keyword gives following lint error: `Data location can only be specified for array, struct or mapping types, but "memory" was given.`

Comment (slice-amandata) [reply contracts/ApplicationRegistry.sol]: toh store mat kar directly use karna.because it will increase the gas fee.@vandan

**Review by slice-amandata — CHANGES_REQUESTED** [review]

**Pull request merged**